src/Apps/Order/Routes/Shipping/index.tsx

```diff
@@ -29,15 +33,91 @@ export interface ShippingProps {
   mediator?: {
     trigger: (action: string, config: object) => void
   }
+  relay?: RelayProp
+}
+
+// TODO: When the todo for abstracting the address is done and we have an Address component, we won't need this here, so the wonky state generic on ShippingRoute is fine for now.
+export interface Address {
+  // TODO: Shipping name is not in Exchange's schema yet.
+  // See: https://artsyproduct.atlassian.net/browse/PURCHASE-377
+  name?: string
+  addressLine1?: string
+  addressLine2?: string
+  city?: string
+  region?: string
+  country?: string
+  postalCode?: string
 }
 
 export interface ShippingState {
   shippingOption: string
 }
 
-export class ShippingRoute extends Component<ShippingProps, ShippingState> {
+export class ShippingRoute extends Component<
+  ShippingProps,
+  ShippingState & Address
+> {
+  // TODO: Fill in with Relay data on load.
+  // See: https://artsyproduct.atlassian.net/browse/PURCHASE-376
   state = {
     shippingOption: "SHIP",
+    address: {},
+  } as ShippingState & Address
+
+  // TODO: This can be handled with Formik.
+  // See: https://artsyproduct.atlassian.net/browse/PURCHASE-375
+  onUpdateName = e => {
+    this.setState({ name: e.target.value })
+  }
+  onUpdateAddressLine1 = e => {
+    this.setState({ addressLine1: e.target.value })
+  }
+  onUpdateAddressLine2 = e => {
+    this.setState({ addressLine2: e.target.value })
+  }
+  onUpdateCity = e => {
```

Comment:
the `{` and `}` are optional here, unless you *really* care about not returning the same value as this.setState

Reply:
Oh yeah, those are left over from an earlier draft of moving the address form into its own component, before I decided to just get the Relay mutation working first. I’ll take them out.